pkg/daemon/cleanup/cleanup_test.go

```diff
@@ -97,6 +97,46 @@ func TestRemoveServices(t *testing.T) {
 	}
 }
 
+func TestMergeServices(t *testing.T) {
+	testIO := []struct {
+		name     string
+		list1    []servicescm.Service
+		list2    []servicescm.Service
+		expected []servicescm.Service
+	}{
+		{
+			name:     "both empty",
+			list1:    []servicescm.Service{},
+			list2:    []servicescm.Service{},
+			expected: []servicescm.Service{},
+		},
+		{
+			name:     "one empty",
+			list1:    []servicescm.Service{{Name: "service1"}, {Name: "service2"}},
+			list2:    []servicescm.Service{},
+			expected: []servicescm.Service{{Name: "service1"}, {Name: "service2"}},
+		},
+		{
+			name:  "different services",
```

Comment:
What happens if the dependency changes for the same service present in both ConfigMaps?

Reply:
looks like list1 will be the source of truth in that case, which in the source code is the most recently created ConfigMap

Reply:
That is obvious. But what will happen in that scenario? For instance, will the incorrect dependency be stopped?